lib/ansible/modules/windows/win_chocolatey.ps1

```diff
@@ -16,532 +19,543 @@ $params = Parse-Args $args -supports_check_mode $true
-            $security_protcols = $security_protcols -bor [Net.SecurityProtocolType]::Tls12
-        }
-        [Net.ServicePointManager]::SecurityProtocol = $security_protcols
-
-        $wc = New-Object System.Net.WebClient;
-        if ($proxy_url)
-        {
-            #We need to configure proxy
-            $env:chocolateyProxyLocation = $proxy_url
-            $wp = New-Object System.Net.WebProxy($proxy_url, $true);
-            $wc.proxy = $wp;
-            if ($proxy_username -and $proxy_password) {
-                $env:chocolateyProxyUser = $proxy_username
-                $env:chocolateyProxyPassword = $proxy_password
-                $passwd = ConvertTo-SecureString $proxy_password -AsPlainText -Force
-                $wp.Credentials = New-Object System.Management.Automation.PSCredential($proxy_username, $passwd)
-            }
-        }
-        $install_output = $wc.DownloadString("https://chocolatey.org/install.ps1") | powershell -
-        $result.rc = $LastExitCode
-        $result.stdout = $install_output | Out-String
-        if ($result.rc -ne 0) {
-            Fail-Json -obj $result -message "Chocolatey bootstrap installation failed."
-        }
-        $result.changed = $true
-        Add-Warning -obj $result -message "Chocolatey was missing from this system, so it was installed during this task run."
-
-        # locate the newly installed choco.exe
-        $command = Get-Command -Name "choco.exe" -ErrorAction SilentlyContinue
-        if ($command)
-        {
-            $path = $command.Path
-        }
-        else
-        {
-            $env_value = $env:ChocolateyInstall
-            if ($env_value)
-            {
-                $path = "$env_value\bin\choco.exe"
-            }
-            else
-            {
-                $path = "$env:SYSTEMDRIVE\ProgramData\Chocolatey\bin\choco.exe"
-            }
-        }
-        if (-not (Test-Path -Path $path))
-        {
-            Fail-Json -obj $result -message "failed to find choco.exe, make sure it is added to the PATH or the env var ChocolateyInstall is set"
-        }
-
-        $script:executable = $path
-    }
-    else
-    {
-
-        $script:executable = "choco.exe"
-
-        if ([Version](choco --version) -lt [Version]'0.10.5')
-        {
-            Add-Warning -obj $result -message "Chocolatey was older than v0.10.5, so it was upgraded during this task run."
-            $script:options = @( "-dv" )
-            Choco-Upgrade -package chocolatey  -proxy_url $proxy_url -proxy_username $proxy_username -proxy_password $proxy_password
-        }
+Function Get-CommonChocolateyArguments {
+    # uses global vars like check_mode and verbosity to control the common args
+    # run with Chocolatey
+    $arguments = [System.Collections.ArrayList]@("--yes", "--no-progress")
+    # global vars that control the arguments
+    if ($check_mode) {
+        $arguments.Add("--what-if") > $null
```

Comment:
I'm using an `ArrayList` for better performance and the [Add](https://msdn.microsoft.com/en-us/library/system.collections.arraylist.add.aspx) returns the index of the added method which I need to redirect to `$null`.

Reply:
Hmm, surprising things... :-)

Reply:
Hi,
may I suggest using 
`$arguments.Add("--debug") | out-null` instead of `$arguments.Add("--debug") > $null`.
I think it's more "Powershell Way".

Thanks

Reply:
@Gianlu `| out-null` is a more expensive operation then casting to `[void]` or redirecting it directly to `$null` https://stackoverflow.com/questions/5260125/whats-the-better-cleaner-way-to-ignore-output-in-powershell. It’s not going to make a noticeable impact really but just something I recommend anyway in case you do deal with lots of these calls.

Pretty much the same reason why I’m using an ArrayList over a generic array.